addons/mail/models/mail_message.py

```diff
@@ -866,11 +868,12 @@ def _message_reaction(self, content, action):
             "message": Store.one_id(self),
         }
         # sudo: mail.guest - guest can send their own name when reacting
-        self._bus_send_store(
-            Store(self, {"reactions": [(group_command, group_values)]}).add(
+        store = Store(self, {"reactions": [(group_command, group_values)]}).add(
                 guest.sudo() or partner, fields=["name", "write_date"]
             )
-        )
+        self._bus_send_store(store)
+        # return reaction data for non subscribed users
+        return store.get_result()
```

Comment:
`store.get_result()` should never be returned in the middle of business code.
It's only for rpc result or for payload of bus.

The pattern is that if necessary the method should should receive a store as param and fill it as necessary.